src/main/scala/mesosphere/marathon/api/akkahttp/EntityMarshallers.scala

```diff
@@ -95,10 +96,46 @@ object EntityMarshallers {
     jsonStringMarshaller.compose(printer).compose(internal => writes.writes(ramlWrites.write(internal)))
 
   import mesosphere.marathon.raml.AppConversion.appRamlReader
-  implicit def appUnmarshaller(
+  implicit def appDefinitionUnmarshaller(
     implicit
-    normalization: Normalization[raml.App], validator: Validator[AppDefinition]) = {
-    validEntityRaml(playJsonUnmarshaller[raml.App])
+    normalization: Normalization[raml.App],
+    validator: Validator[AppDefinition]): FromEntityUnmarshaller[AppDefinition] = {
+    validEntityRaml(playJsonUnmarshaller[raml.App]).handleValidationErrors
+  }
+
+  implicit val appDefinitionMarshaller: ToEntityMarshaller[AppDefinition] = playJsonMarshaller[AppDefinition]
```

Comment:
Does this implicitly convert to raml, and then to json? Might be clearer to use a different method name?

Maybe something like:

```
ramlJsonMarshaller[raml.App, AppDefinition]
```

?

Reply:
I've used the same for the info endpoint. However, it should be `implicit val appDefinitionMarshaller: ToEntityMarshaller[raml.App] = playJsonMarshaller[raml.App]` if I'm not mistaken.

Reply:
`internalToRamlJsonMarshaller[raml.App, AppDefinition]` doesn't compile for some reason. I'd just wait to use it in AppsController and then refactor it properly